db.go

```diff
@@ -661,14 +692,24 @@ func (db *DB) Get(dest interface{}, keys ...interface{}) error {
 // Returns an error if an element in the list has not been registered
 // with BindModel.
 func (db *DB) Insert(list ...interface{}) error {
-	return insertObjects(db, db, getInsert, list)
+	return insertObjects(db.Context(), db, db, getInsert, list)
+}
+
+// InsertContext is the context version of Insert.
+func (db *DB) InsertContext(ctx context.Context, list ...interface{}) error {
+	return insertObjects(ctx, db, db, getInsert, list)
 }
 
 // Query executes a query that returns rows, typically a SELECT. The
 // args are for any placeholder parameters in the query. This is a
 // small wrapper around sql.DB.Query that returns a *squalor.Rows
 // instead.
 func (db *DB) Query(q interface{}, args ...interface{}) (*Rows, error) {
+	return db.QueryContext(db.Context(), q, args...)
+}
+
+// QueryContext is the context version of Query.
+func (db *DB) QueryContext(ctx context.Context, q interface{}, args ...interface{}) (*Rows, error) {
 	start := time.Now()
 	if db.context != nil {
```

Comment:
`ctx` instead of `db.context`

Reply:
Fixed, and changed to `isEmptyContext(ctx)`